### PR TITLE
feat: create basic logic and ui

### PR DESCRIPTION
**Pull request by lewxdev**
~~ran into a roadblock with the interactivity, need to look more into `hono/jsx/dom`~~

we have interactivity! I can add the rest of the logic later

mmmines-ui

### CODE REVIEW

File: src/client.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/// <reference lib="dom" />
```
Comment (drewradcliff) [inline src/client.tsx:1]: would be nice to not have to do this

File: src/game.ts
Comment (drewradcliff) [inline src/game.ts:1]: It might be nice to have this generation from an API endpoint like `/seed` to generate the board at some point

**Review by drewradcliff — APPROVED** [review]
Looks good. I'm sure there's going to be a lot of refactoring of the game logic as we seed and query the redis store

File: src/client.tsx (hunk 1 above)
Comment (lewxdev) [reply src/client.tsx:1]: the alternative is moving this to the `tsconfig.json`
```jsonc
{
  // ...
  "lib": ["ESNext", "DOM"],
  // ...
}
```
this comes at the risk of mistaking the existence of DOM APIs in server code. tbh I think using the directive as we currently have it here is more type safe

File: src/game.ts
Comment (lewxdev) [reply src/game.ts:1]: later yeah

Comment (lewxdev) [reply src/game.ts:1]: I guess later is now 👁️ 👄 👁️

**Pull request merged**